### DISCUSSION TITLE
prioritization of issues in Arctos

### DISCUSSION THREAD

**Discussion by catherpes** [ArctosDB/arctos#6207] [General]
How are issues prioritized in Arctos?

I'm really frustrated right now. I've filed an issue weeks ago regarding the burdensome multi-step login sequence, and all of the energy of this community at this moment is being used on cleaning up what apparently is a legacy attribute field of right gonad width??

Can we please make it a priority to work on things that the most important users of Arctos, the Collection Staff, whose collections budgets pay money to work every day in this frustrating environment, have identified as things that would improve their user experience, rather than things identified by the programmers that could sit for DECADES and nobody would really care about? 

If there's something more that I need to do regarding this issue I filed, please let me know, but I don't remember anything being asked of me regarding this.

**Comment by dustymc** [comment]
I believe you're referring to https://github.com/ArctosDB/arctos/issues/5852. I did some stuff, I don't know how to do more stuff.

https://github.com/ArctosDB/internal/issues/258 is still being sorted out, but it is my usual plea for a way to do just what you're asking (along with not doing things that then get immediately un-done because apparently nobody was paying attention): prioritize the things that need prioritizing, don't get distracted by the things that don't need priority. The first step is to get something actionable, and as above I need help on that. I'm up for most anything regarding how the mechanics of any of that work - use the issue, show up to one of the office hour things, show up to most any scheduled meeting, schedule something new, send smoke signals, WHATEVER, I'll try to make it work (and I think everyone else feels the same way).

**Comment by Jegelewicz** [comment]
@catherpes issues with a lot of participation by the community get priority. The more users affected, the higher the priority. Also those that affect data quality (the right gonad width issue does) or are bugs (Arctos is not working). I understand your frustration, but nobody else in the community seems to have issues with the login process or they would have added to it. In the past, things have been changed based on a single user's request that have come back to haunt the community. If the login issue is very important to you, I suggest adding to the AWG Issues agenda and attending the meeting to plead your case or recruit other users from your institution to comment on the issue.

**Comment by campmlc** [comment]
I have similar issues with our UI, many of which I have posted and which
have not been resolved. This is not because these issues aren't valid - but
most people in our user community are just trying to do their jobs and
don't have time to post and follow issues. In the absence of a UI
developer, this must be a priority role of the AWG and officers - to make
the user environment functional. I suggest we continue our UI development
as a standing committee meeting. I will send out a poll to schedule.

**Comment by campmlc** [comment]
@mkoo

**Comment by catherpes** [comment]
This is ridiculous. This feels like stonewalling. Recruit supporters? 
I'm not a politician. I have a job to do. We as collection managers all 
have jobs to do that aren't database development. Arctos should be 
helping us do our jobs. Give me something to try out. I didn't see any 
request for more information 5852.

I'm beyond frustrated, downright angry now. Taking a break from this 
before I say something I regret.

**Comment by catherpes** [comment]
There are 4.8 M records in Artos and right gonad width affects 226 records

Is that a good use of resources???

**Comment by mkoo** [comment]
@catherpes Appreciate your thoughts. First I wanted to say generally the use of github issues is for a community input and action. A lot of the detailed work can be done in working groups or taken up by one of our monthly meetings. "Hot" topics does not mean it's a priority for development -- That said, the github forum is handy for weighing in on a lot of conversations and it can be overwhelming. We always recommend that people tweak their personal preference for what works for you (eg turn off all notifications but just receive specific ones, get a digest, use email filters). And pay attention to Labels and Milestones. Some issues are filed specifically for discussion! 

As for development prioritization, we will be discussing and introducing at the next monthly Working Group meeting (4th Thursdays at 11 am PT) some new measures to make this hopefully more transparent. 

OK, now to some specifics:[ Streamlining login](https://github.com/ArctosDB/arctos/issues/5852) Seems like there are some incremental things that Dusty is doing. Password managers definitely streamline and obfuscate the need to address for many users so this is a difficult one to suss out unless we see it in action. (for ex., I get logged in immediately after clicking log in, and just close the window (red X in the upper right) reflexively; BUT sometimes I do go straight to notifications because I have to fish out an error or IP address). AND I think we can work on the re-logging in after time-out issue IMO. Regardless, we wont close until we're good with all aspects. I also added to UI project so that the UI working group can consider.

As for right/left/ unknown[ gonad measurements](https://github.com/ArctosDB/arctos/issues/6189), yes now it affects few records but there is an active NSF grant to record traits so I imagine this is better to sort out now when it affects few before it becomes another "hot and urgent" topic and they are trying to load many. We do need a space for discussions and if the diverse viewpoints can be gathered before the Code Table committee convenes all the better (plus the legacy CT just needs cleaning up so these tend to be constantly brought up). 

To anyone reading: Everything Teresa said BTW applies but I hope I added a little more detail. I personally am grateful when people get involved constructively on issues they are passionate about (for whatever work or research reasons) and speaking for the Arctos officers, we appreciate people understanding that we are similarly all working at our respective museums with only two fulltime Arctos staff, a handful of Arctos volunteer officers, and a committed collection community. Feel free to reach out to us directly too anytime: Arctos Working Group Officers <arctos-working-group-officers@googlegroups.com>

**Comment by catherpes** [comment]
Thanks Michelle.

I'll lay out my perspective as constructively as I can here one last 
time because I don't think I've ever posted a version of this email to 
github, although individuals have received similar versions over the 
years. Some I wrote and never sent. Call it a rant if you must.

I've been trying to participate here on github lately when things really 
seemed to be accelerating off the rails with respect to institutional 
catalog numbers, or in the past when male? and female? was proposed for 
elimination but I don't see myself getting involved more than at that 
level because it is an emotional burden that I don't need in my life. 
I'm not inclined to spend any more time than I already do participating 
in these generally frustrating discussions. Decision by committee is 
painfully inefficient, especially when everyone but two people involved 
has another full-time job to do, and that seems to be all that Arctos 
does - decision by committee.

MSB Birds got in to Arctos shortly after I got here nineteen years ago 
and I've never found functionality for CMs to be a top priority for the 
Database. The strategy from Joe and Gordon was to get as many 
collections in so it's too big to fail, get functionality unique to this 
platfrom (linking Arctos records to Genbank, pubs, projects, etc., etc.) 
for the public user and to impress potential funding sources, with no 
regard for how my colleagues and I were supposed to get the records in 
efficiently. 'email Dusty' Joe would say. Then, 'put it on github' I'd 
be told. Gordon once stood in my office door, conceded that all of my 
complaints of data entry and record editing from a CM's perspective were 
valid, and told me we're all beta testers. There's some dark humor. Now 
Arctos is millions of records in and I'm more frustrated than I've ever 
been.

Last month, I had to get Mariel to finish the bulkload of the most 
recent thousand-plus MSB Bird records into Arctos. I've never had to do 
that before (get someone to finish the process for me), and the file I 
sent her was version 6. That means I'd already done six iterations of 
formatting the file, finding errors by getting it into the staging 
table, trying to bulkload it, and getting the error file back. It was 
close, I thought, but it still took her three weeks to get it done. 
There were loan requests to be filled that were dependent on getting 
these records into the database, so it was a priority for her. And it 
still took her three weeks. The data entry interface, revamped though it 
is, still is a mouse-intensive popup-clicking exercise that makes the 
process inefficient and frustrating to the point that bulkloading is 
still my best option.

That's great that there's a grant to work on attributes/traits related 
to repro data. I'd love to hear of a dedicated funding source to work on 
something so mundane but fundamental as data entry and editing.

It's my opinion that the developers of Arctos are too close to it. It's 
like a mural on the side of a sky scraper being painted one brick at a 
time and nobody is asking the critics viewing from across the parking 
lot to see if this masterpiece actually is looking like it's supposed 
to. Do the colors and values actually work together? In some places yes, 
but in others not at all. The public, viewing from afar, sees a slightly 
abstract representation that they're fairly pleased with, but they don't 
know that it's actually supposed to be a realistic representation of the 
subject matter.

Has there ever been an outside audit of the functionality of this 
database? Somebody from a computer science background who knows about 
database user friendliness? How to make it more efficient from the 
standpoint of the most important users, the CMs? Our data would be 
better and available more quickly if Arctos were more usable, and that's 
ultimately what this is all about: Serving data to the public.

That's where I'm at. I've resigned myself to doing the minimum in Arctos 
that my job requires: getting records into arctos and getting the tissue 
part barcodes entered so that they can be used in loans. Few media, and 
most projects and links to pubs were not entered by me. I work in it 
sparingly because I find it incredibly frustrating.

I filed login issue 5852 because Arctos login is a convoluted process 
that I've not experienced in any other login interface on the web. Only 
logins that require secondary authentication take more time, but at 
least I understand why the process works the way it does, so they are 
less annoying. I thought that surely this little nugget of functionality 
wouldn't be too hard implement! I tried to make specific critiques and 
suggestions to fix them. However, it has become another representative 
microcosm of what frustrates me about this program.

Thanks for your attention.

Andy

**Comment by mkoo** [comment]
thanks again Andy.  I will migrate this to discussions and close as needed. I feel this might more fruitful if this is in realtime. Let me know if we can chat
